react-native.js

```diff
@@ -7,36 +7,45 @@ const ERROR = 2;
 module.exports = {
   extends: [
     require.resolve('./node.js'),
-    'plugin:react/recommended',
-    'prettier/react',
+    require.resolve('./react.js'),
     'plugin:react-native-a11y/all',
   ],
   env: {
     'react-native/react-native': true,
   },
-  plugins: ['react', 'react-native', 'react-hooks'],
-  parserOptions: {
-    ecmaFeatures: {
-      jsx: true,
-    },
-  },
+  plugins: ['react-native'],
   rules: {
-    'react/display-name': OFF,
-    'react/no-multi-comp': [WARNING, { ignoreStateless: true }],
-    'react/no-unused-prop-types': OFF,
-    'react/prop-types': OFF,
-    'react/require-default-props': OFF,
     'react-native/no-unused-styles': ERROR,
     'react-native/split-platform-components': OFF,
     'react-native/no-inline-styles': WARNING,
     'react-native/no-color-literals': WARNING,
     'react-native/no-raw-text': ERROR,
-    'react-hooks/rules-of-hooks': ERROR,
-    'react-hooks/exhaustive-deps': WARNING,
   },
-  settings: {
-    react: {
-      version: 'detect',
+  overrides: [
+    {
+      files: ['*.js', '*.jsx'],
+      settings: {
+        'import/extensions': extensions.JS_REACT_NATIVE,
+        'import/resolver': {
+          node: {
+            extensions: extensions.JS_REACT_NATIVE,
+          },
+        },
+      },
     },
-  },
+    {
+      files: ['*.ts', '*.tsx'],
+      settings: {
+        'import/extensions': extensions.TS_REACT_NATIVE,
```

Comment:
Same question as for JS_REACT_NATIVE above: should .{ts,tsx,js,jsx} extension be also included here?